guides/keras_hub/object_detection_retinanet.py

```diff
@@ -0,0 +1,796 @@
+
+    Args:
+      record: A dictionary representing a single TFDS record.
+
+    Returns:
+      A dictionary with "images" and "bounding_boxes".
+    """
+    image = record["image"]
+    image_shape = tf.shape(image)
+    height, width = image_shape[0], image_shape[1]
+    boxes = keras.utils.bounding_boxes.convert_format(
+        record["objects"]["bbox"],
+        source="rel_yxyx",
+        target=gt_bbox_format,
+        height=height,
+        width=width,
+    )
+    labels = record["objects"]["label"]
+
+    bounding_boxes = {"boxes": boxes, "labels": labels}
+
+    return {"images": image, "bounding_boxes": bounding_boxes}
+
+
+def preprocess_tfds(ds: tf.data.Dataset) -> tf.data.Dataset:
+    """Preprocesses a TFDS dataset for object detection.
+
+    Args:
+        ds: The TFDS dataset.
+        resizing: A resizing function.
+        max_box_layer: A max box processing function.
+        batch_size: The batch size.
+
+    Returns:
+      A preprocessed TFDS dataset.
+    """
+    ds = ds.map(resizing, num_parallel_calls=tf.data.AUTOTUNE)
+    ds = ds.map(max_box_layer, num_parallel_calls=tf.data.AUTOTUNE)
+    ds = ds.batch(batch_size, drop_remainder=True)
+    return ds
+
+
+"""
+Now concatenate both 2007 and 2012 VOC data
+"""
+train_ds = train_ds_2007.concatenate(train_ds_2012)
+train_ds = train_ds.map(decode_custom_tfds, num_parallel_calls=tf.data.AUTOTUNE)
+train_ds = preprocess_tfds(train_ds)
+
+"""
+Load the eval data
+"""
+eval_ds = tfds.load("voc/2007", split="test")
+eval_ds = eval_ds.map(decode_tfds, num_parallel_calls=tf.data.AUTOTUNE)
+eval_ds = preprocess_tfds(eval_ds)
+
+"""
+### Let's visualize batch of training data
+"""
+record = next(iter(train_ds.shuffle(100).take(1)))
+keras.visualization.plot_bounding_box_gallery(
+    record["images"],
+    bounding_box_format=gt_bbox_format,
+    y_true=record["bounding_boxes"],
+    scale=3,
+    rows=2,
+    cols=2,
+    class_mapping=INDEX_TO_CLASS,
+)
+
+"""
+### Decoded TFDS record to a tuple for keras-hub
+"""
+train_ds = train_ds.map(convert_to_tuple, num_parallel_calls=tf.data.AUTOTUNE)
+train_ds = train_ds.prefetch(tf.data.AUTOTUNE)
+
+eval_ds = eval_ds.map(convert_to_tuple, num_parallel_calls=tf.data.AUTOTUNE)
+eval_ds = eval_ds.prefetch(tf.data.AUTOTUNE)
+
+"""
+## Configure RetinaNet Model
+Configure the model with `backbone`, `num_classes` and `preprocessor`.
+Use callbacks for recording logs and saving checkpoints.
+"""
+
+
+def get_callbacks(experiment_path: str):
+    """Creates a list of callbacks for model training.
+
+    Args:
+      experiment_path (str): Path to the experiment directory.
+
+    Returns:
+      List of keras callback instances.
+    """
+    # models_path = os.path.join(experiment_path, "tf_models")
+    tb_logs_path = os.path.join(experiment_path, "logs")
+    ckpt_path = os.path.join(experiment_path, "weights")
+    return [
+        keras.callbacks.BackupAndRestore(ckpt_path, delete_checkpoint=False),
```

Comment:
are we using the `BackupAndRestore` or `TensorBoard` callbacks in this guide anywhere? if not than maybe cleaner to remove

Reply:
This was used in case if user wants to do any custom training and wants to use callbacks.